### PR TITLE
Make craft without arguments craft as many items as possible.

### LINKED ISSUE #806: Craft without arguments doesn't craft as many items as possible
The help says that craft without any arguments will craft as many items as possible, but it just shows the command usage.

### PR DESCRIPTION
**Pull request by fatboychummy**
Fixes #806

### CODE REVIEW

**Comment by fatboychummy** [comment]
I'm unable to test the tests currently -- It's quite late, I don't have gradle on this pc yet, and I need to do the sleep thing.

I'm assuming the tests will run in the pr build though, if anything fails I can fix it when I wake up.

**Review by SquidDev — CHANGES_REQUESTED** [review]
Sorry, should have left a comment on the original issue about this: I think a better solution would be that `craft` still displays its usage, but `craft all` crafts as many items as possible. Obviously (and ideally auto-complete) would have to update the docs to reflect this too.

The rationale here being that `craft` can be a destructive action at times, and I'd rather than shell commands didn't do too much damage when run without args.

File: src/main/resources/data/computercraft/lua/rom/programs/turtle/craft.lua [hunk 1]
```diff
@@ -11,9 +11,9 @@ end
 local tArgs = { ... }
 local nLimit = tonumber(tArgs[1])
 
-if not nLimit and tArgs[1] then
+if not nLimit and tArgs[1] ~= "all" then
     local programName = arg[0] or fs.getName(shell.getRunningProgram())
-    print("Usage: " .. programName .. " [number]")
+    print("Usage: " .. programName .. " [all/number]")
```
Comment (SquidDev) [inline src/main/resources/data/computercraft/lua/rom/programs/turtle/craft.lua:16]: As the amount is required, it should be `<number>` now - I think `all|<number>` makes the most sense, or have two separate usage lines.

**Review by SquidDev — CHANGES_REQUESTED** [review]
Thank you! Just one minor nitpick, but otherwise looks good!

File: src/main/resources/data/computercraft/lua/rom/programs/turtle/craft.lua (hunk 1 above)
Comment (fatboychummy) [reply src/main/resources/data/computercraft/lua/rom/programs/turtle/craft.lua]: Alright, it's now using `all|<number>`

**Review by SquidDev — APPROVED** [review]

**Comment by SquidDev** [comment]
Thank you!

**Pull request merged**